src/samplers/multinomial.jl

```diff
@@ -1,33 +1,33 @@
-function multinom_rand!(rng::AbstractRNG, n::Int, p::AbstractVector{Float64},
-                         x::AbstractVector{<:Real})
+function multinom_rand!(rng::AbstractRNG, n::Int, p::AbstractVector{Tp},
+                         x::AbstractVector{Tx}) where {Tx <: Real, Tp <: Real}
     k = length(p)
     length(x) == k || throw(DimensionMismatch("Invalid argument dimension."))
 
-    rp = 1.0  # remaining total probability
+    rp = Tp(1.0)  # remaining total probability
```

Comment:
I'd suggest something like
```suggestion
    z = zero(eltype(p))
    rp = oftype(z + z, 1) # remaining total probability (widens type if needed)
```
This will also work better with `p::Vector{Bool}`.